### PR TITLE
JBoss SQLi

### PR DESCRIPTION
**Pull request by chgg-kboberg**
add SQLi detection for Java services using JBoss Hibernate for database connection management.

### CODE REVIEW

**Review by colleend — APPROVED** [review]
lgtm!! this is awesome!

File: java/jboss/security/session_sqli.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+import org.apache.commons.io.FilenameUtils;
+
+public class Cls extends HttpServlet
+{
+	private static org.apache.log4j.Logger log = Logger.getLogger(Register.class);
+
+    protected void danger(String ean) {
```
Comment (mschwager) [inline java/jboss/security/session_sqli.java:20]:
```suggestion
    // ruleid:find-sql-string-concatenation
    protected void danger(String ean) {
```

File: java/jboss/security/session_sqli.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            while (rs.next()) {
+                Integer item = rs.getInt("foo");
+            }
+        } catch (SQLException e) {
+            logger.error("Error!", e);
+        } finally {
+            session.close();
+        }
+    }
+    protected void danger2(String biz) {
```
Comment (mschwager) [inline java/jboss/security/session_sqli.java:37]:
```suggestion
    // ruleid:find-sql-string-concatenation
    protected void danger2(String biz) {
```

File: java/jboss/security/session_sqli.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+                Integer item = rs.getInt("foo");
+            }
+        } catch (SQLException e) {
+            logger.error("Error!", e);
+        } finally {
+            session.close();
+        }
+    }
+
+    protected void ok(String foo) throws ServletException, IOException {
```
Comment (mschwager) [inline java/jboss/security/session_sqli.java:54]:
```suggestion
    // ok:find-sql-string-concatenation
    protected void ok(String foo) throws ServletException, IOException {
```

File: java/jboss/security/session_sqli.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
+rules:
+- id: find-sql-string-concatenation
+  patterns:
+    - pattern-either: 
+      - pattern: |
+          public $RETURN $METHOD(...,String $X,...){
```
Comment (mschwager) [inline java/jboss/security/session_sqli.yaml]:
```suggestion
          $RETURN $METHOD(...,String $X,...){
```

File: java/jboss/security/session_sqli.yaml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+            ...
+            String $QUERY = ... + $X + ...;
+            ...
+            PreparedStatement $PS = $SESSION.connection().prepareStatement($QUERY);
+            ...
+            ResultSet $RESULT = $PS.executeQuery();
+            ...
+          }
+      - pattern: |
+          public $RETURN $METHOD(...,String $X,...){
```
Comment (mschwager) [inline java/jboss/security/session_sqli.yaml]:
```suggestion
          $RETURN $METHOD(...,String $X,...){
```

File: java/jboss/security/session_sqli.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+public class Cls extends HttpServlet
+{
+	private static org.apache.log4j.Logger log = Logger.getLogger(Register.class);
+
+    // ruleid:find-sql-string-concatenation
+    protected void danger(String ean) {
+        Session session = this.sessionFactory.openSession();
+
+        String query = "select foo from bar where" + biz + " limit 1";
```
Comment (mschwager) [inline java/jboss/security/session_sqli.java]:
```suggestion
        String query = "select foo from bar where" + ean + " limit 1";
```

**Comment by mschwager** [comment]
Thanks again!

**Pull request merged**